src/http/server.clj

```diff
@@ -337,6 +354,7 @@
       (let [port (if (not (empty? arguments))
                    (Integer/parseInt (first arguments))
                    (:port options))
+            basic-auth (when (true? (:auth options)) (auth-prompt))
```

Comment:
I think it's more convenient to run the server with auth params provided (at least a user name). So it should handle 

```
-m http.server --auth username:password
```

or 

```
-m http.server --auth username
```

prompting the password.

Reply:
Totally agree - I was originally going down that path but when I re-read #12 "the server should prompt the user to provide user/password" I thought you wanted a prompt to pop up.  I'll change this later tonight or tomorrow to reflect your recommendations above!